source/patterns/@aws-solutions-constructs/aws-lambda-eventbridge/test/aws-lambda-eventbridge.test.ts

```diff
@@ -0,0 +1,508 @@
+        }
+      ]
+    }
+  });
+});
+
+// --------------------------------------------------------------
+// Test minimal deployment with an existing VPC and existing Lambda function not in a VPC
+//
+// buildLambdaFunction should throw an error if the Lambda function is not
+// attached to a VPC
+// --------------------------------------------------------------
+test("Test minimal deployment with an existing VPC and existing Lambda function not in a VPC", () => {
+  // Stack
+  const stack = new Stack();
+
+  const testLambdaFunction = new lambda.Function(stack, 'test-lamba', {
+    runtime: lambda.Runtime.NODEJS_12_X,
+    handler: "index.handler",
+    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+  });
+
+  const testVpc = new ec2.Vpc(stack, "test-vpc", {});
+
+  // Helper declaration
+  const app = () => {
+    // Helper declaration
+    new LambdaToEventbridge(stack, "lambda-to-eventbridge-stack", {
+      existingLambdaObj: testLambdaFunction,
+      existingVpc: testVpc,
+    });
+  };
+
+  // Assertion
+  expect(app).toThrowError();
+
+});
+
+// ----------------------------------------------------------------------------------
+// Test lambda function custom environment variable and custom event bus permissions
+// ----------------------------------------------------------------------------------
+test('Test lambda function custom environment variable', () => {
+  // Stack
+  const stack = new Stack();
+
+  // Helper declaration
+  new LambdaToEventbridge(stack, 'lambda-to-eventbridge-stack', {
+    lambdaFunctionProps: {
+      runtime: lambda.Runtime.NODEJS_12_X,
+      handler: 'index.handler',
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    },
+    eventBusProps: {eventBusName: 'customeventbus'},
+    eventBusEnvironmentVariableName: 'CUSTOM_EVENTBUS_NAME'
+  });
+
+  // Check environment variables
+  expect(stack).toHaveResource('AWS::Lambda::Function', {
+    Handler: 'index.handler',
+    Runtime: 'nodejs12.x',
+    Environment: {
+      Variables: {
+        AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
+        CUSTOM_EVENTBUS_NAME: {
+          Ref: "lambdatoeventbridgestackcustomeventbus25825BEF"
+        }
+      }
+    }
+  });
+
+  expect(stack).toHaveResource('AWS::Events::EventBus', {
+    Name: `customeventbus`
+  });
+
+  // Check lambda permissions to custom event bus
+  expect(stack).toHaveResourceLike("AWS::IAM::Policy", {
+    PolicyDocument: {
+      Statement: [
+        {
+          Action: [
+            "xray:PutTraceSegments",
+            "xray:PutTelemetryRecords"
+          ],
+          Effect: "Allow",
+          Resource: "*"
+        },
+        {
+          Action: "events:PutEvents",
+          Effect: "Allow",
+          Resource: {
+            "Fn::GetAtt": [
+              "lambdatoeventbridgestackcustomeventbus25825BEF",
+              "Arn"
+            ]
+          }
+        }
+      ]
+    }
+  });
+});
+
+test('check multiple constructs in a single stack', () => {
```

Comment:
How about a test that shares 1 event bus between 2 Lambda functions (create it with the first construct, share it in the second construct)

Reply:
added the test